src/Chapter3.hs

```diff
@@ -480,6 +516,37 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city in total.
 -}
 
+
+type CastleName = String
+type Walls = Int
+newtype House = House Int
```

Comment:
To avoid building houses with more than 4 people you need to restrict usage of the Constructor and allow to create `House`s only with, for example,  `mkHouse` function.

You can read a bit more on the pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#smart-constructor

Generally, in here, you could restrict that on type level, by using something else than `Int` 🙂